core/src/test/java/org/apache/iceberg/TestSnapshotManager.java

```diff
@@ -250,4 +248,305 @@ public void testCherryPickOverwrite() {
+    long snapshotId = table.currentSnapshot().snapshotId();
+    // Test a basic case of creating a branch
+    table.manageSnapshots()
+        .createBranch("branch1", snapshotId)
+        .commit();
+    SnapshotRef expectedBranch = table.ops().refresh().ref("branch1");
+    Assert.assertTrue(expectedBranch != null &&
+            expectedBranch.equals(SnapshotRef.branchBuilder(snapshotId).build()));
+
+    // Trying to create a branch with an existing name should fail
+    AssertHelpers.assertThrows("Should fail validation check if we try and create branch again",
+        IllegalArgumentException.class, "Ref branch1 already exists",
+        () -> table.manageSnapshots().createBranch("branch1", snapshotId).commit());
+
+    // Trying to create another branch within the same chain
+    AssertHelpers.assertThrows("Should fail validation check if we try and create branch again",
+        IllegalArgumentException.class, "Ref branch2 already exists",
+        () -> table.manageSnapshots().createBranch("branch2", snapshotId).createBranch("branch2", snapshotId).commit());
+  }
+
+
+  @Test
+  public void testCreateTag() {
+    table.newAppend()
+        .appendFile(FILE_A)
+        .commit();
+    long snapshotId = table.currentSnapshot().snapshotId();
+    // Test a basic case of creating a tag
+    table.manageSnapshots()
+        .createTag("tag1", snapshotId)
+        .commit();
+    SnapshotRef expectedTag = table.ops().refresh().ref("tag1");
+
+    Assert.assertTrue(expectedTag != null &&
+        expectedTag.equals(SnapshotRef.tagBuilder(snapshotId).build()));
+
+    // Trying to create a tag with an existing name should fail
+    AssertHelpers.assertThrows("Should fail validation check if we try and create tag again",
+        IllegalArgumentException.class, "Ref tag1 already exists",
+        () -> table.manageSnapshots().createTag("tag1", snapshotId).commit());
+
+    // Trying to create another tag within the same chain
+    AssertHelpers.assertThrows("Should fail validation check if we try and create tag again",
+        IllegalArgumentException.class, "Ref tag2 already exists",
+        () -> table.manageSnapshots()
+            .createBranch("tag2", snapshotId)
+            .createBranch("tag2", snapshotId).commit());
+  }
+
+  @Test
+  public void testRemoveBranch() {
+    table.newAppend()
+        .appendFile(FILE_A)
+        .commit();
+    long snapshotId = table.currentSnapshot().snapshotId();
+    // Test a basic case of creating and then removing a branch and tag
+    table.manageSnapshots()
+        .createBranch("branch1", snapshotId)
+        .commit();
+    table.manageSnapshots()
+        .removeBranch("branch1")
+        .commit();
+
+    TableMetadata updated = table.ops().refresh();
+    SnapshotRef expectedBranch = updated.ref("branch1");
+    Assert.assertTrue(expectedBranch == null);
+
+    // Test chained creating and removal of branch and tag
+    table.manageSnapshots()
+        .createBranch("branch2", 1)
```

Comment:
I think this should use `snapshotId` rather than `1`. Otherwise the test is brittle and could break if snapshot ID assignment changes.

Reply:
Updated